docs/connection.md

```diff
@@ -38,16 +38,16 @@ The developer needs to implement four public coroutines:
 
 - The `disconnect` coroutine implements the teardown logic required to be performed for the connection when it is eventually stopped. The `disconnect` coroutine is called by the AEA framework once when the agent is being stopped.
 
-- The `send` coroutine is called by the AEA framework each time when the `Multiplexer` handles an outgoing envelope specified to be handled by the connection. The `send` coroutine must implement the processing of the envelope leaving the agent.
+- The `send` coroutine is called by the AEA framework each time the `Multiplexer` handles an outgoing envelope specified to be handled by this connection. The `send` coroutine must implement the processing of the envelope leaving the agent.
 
-- The `receive` coroutine is continuously called by the AEA framework. It either returns `None` or an envelope. The `receive` coroutine must implement the logic of data being received by the agent and if necessary its translation into a relevant protocol.
+- The `receive` coroutine is continuously called by the AEA framework. It either returns `None` or an envelope. The `receive` coroutine must implement the logic of data being received by the agent, and if necessary, its translation into a relevant protocol.
 
 
-When developing your own connection you might benefit from inspecting the `fetchai/http_server:0.18.0` and `fetchai/http_client:0.19.0` connections to gain more familiarity and inspiration.
+When developing your own connection you might benefit from inspecting the `fetchai/http_server:0.18.0` and `fetchai/http_client:0.19.0` connections to gain more familiarity and inspiration. These connections are for interfacing with http clients and servers respectively.
```

Comment:
Note, the second sentence will be hard for people to understand. Probably best say, http server is for external clients to connect, etc. i.e. collocate explanation